include/deal.II/base/parallel.h

```diff
@@ -542,20 +542,73 @@ namespace parallel
                           const Function                             &f,
                           const unsigned int                          grainsize)
   {
-#ifndef DEAL_II_WITH_TBB
-    // make sure we don't get compiler
-    // warnings about unused arguments
-    (void)grainsize;
-
-    f(begin, end);
-#else
+#ifdef DEAL_II_WITH_TASKFLOW
+    tf::Executor &executor = MultithreadInfo::get_taskflow_executor();
+    tf::Taskflow  taskflow;
+
+    if constexpr (std::is_integral_v<Iterator>)
+      {
+        // Integral "iterator" types (e.g., int): use plain arithmetic
+        using integral_type = Iterator;
+
+        const integral_type           n = end - begin;
+        tf::IndexRange<integral_type> range(0, n, 1);
+
+        taskflow.for_each_by_index(
+          range,
+          [&, begin](tf::IndexRange<integral_type> subrange) {
+            integral_type subrange_begin =
+              begin + static_cast<integral_type>(subrange.begin());
+
+            integral_type subrange_end =
+              subrange_begin + static_cast<integral_type>(subrange.size());
+
+            f(subrange_begin, subrange_end);
+          },
+          tf::GuidedPartitioner(grainsize));
+      }
+    else
+      {
+        // Non-integral iterator types: advance from 'begin' using
```

Comment:
I think if you have iterators instead of indices, you should be using for_each (https://taskflow.github.io/taskflow/classtf_1_1FlowBuilder.html#aae3edfa278baa75b08414e083c14c836)